### PR TITLE
[vcpkg baseline][graphviz/think-cell-range/geotrans] Fix download error

### PR DESCRIPTION
**Pull request by LilyWangLL**
Fix baseline [Pipelines - Run 20230601.1 (azure.com)](https://dev.azure.com/vcpkg/public/_build/results?buildId=90122&view=results) ports build error.

1. Update download hash of `geotrans`.
2. `think-cell-range` build failed due to incorrect hash, update it to new version.
3. `graphviz:arm64-windows` build failed with cannot find file `ltdl.h`, it becauses there is no arm64 folder in download path: https://gitlab.com/graphviz/graphviz-windows-dependencies, so use x64 folder instead.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/geotrans/portfile.cmake [hunk 1]
```diff
@@ -4,8 +4,8 @@ set(VCPKG_LIBRARY_LINKAGE "dynamic")
 # which we re-build anyway.  There is no source only package provided or it would be preferred (and smaller).
 vcpkg_download_distfile(ARCHIVE
     URLS "https://earth-info.nga.mil/php/download.php?file=wgs-mastertgz"
-    FILENAME "geotrans-3.9-master.tgz"
-    SHA512 b0bd5ca6eb584eebf0adc89eb86ec949f372bb4ea92c559f74f2055b0f121369b3d8de0bcff7b7db9abc57ac5a129d5c283d67f098e4af12d91b41747a76e541
+    FILENAME "geotrans-${VERSION}-master.tgz"
```
Comment (dg0yt) [inline ports/geotrans/portfile.cmake:7]: Should we add some kind of disambiguator to the filename, now that the content changed? 
e.g. `geotrans-${VERSION}-master-b0bd5ca.tgz`

File: ports/graphviz/portfile.cmake [hunk 2]
```diff
@@ -24,6 +22,9 @@ endif()
 
 vcpkg_list(SET OPTIONS)
 if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
+    if (VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+        set(VCPKG_TARGET_ARCHITECTURE "x64")
+    endif()
```
[the remark this answers is not in the document]
Comment (LilyWangLL) [reply ports/graphviz/portfile.cmake:27]: > 3. graphviz:arm64-windows build failed with cannot find file ltdl.h, it becauses there is no arm64 folder in download path: https://gitlab.com/graphviz/graphviz-windows-dependencies/-/raw/141d3a21be904fa8dc2ae3ed01d36684db07a35d, so use x64 folder instead.

The reason in description of this PR.

File: ports/geotrans/portfile.cmake (hunk 1 above)
Comment (LilyWangLL) [reply ports/geotrans/portfile.cmake]: Done. Thanks~

File: ports/graphviz/portfile.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/graphviz/portfile.cmake:27]: I do believe that the x64 download works for this download.
But I don't believe that you really want to change the `VCPKG_TARGET_ARCHITECTURE` variable in a portfile.

File: ports/graphviz/portfile.cmake [hunk 3]
```diff
@@ -22,12 +22,15 @@ endif()
...
+    else()
+        set(FOLDER_PATH ${VCPKG_TARGET_ARCHITECTURE})
     endif()
     vcpkg_download_distfile(
         LTDL_H_PATH
-        URLS "https://gitlab.com/graphviz/graphviz-windows-dependencies/-/raw/141d3a21be904fa8dc2ae3ed01d36684db07a35d/${VCPKG_TARGET_ARCHITECTURE}/include/ltdl.h"
+        URLS "https://gitlab.com/graphviz/graphviz-windows-dependencies/-/raw/141d3a21be904fa8dc2ae3ed01d36684db07a35d/${FOLDER_PATH}/include/ltdl.h"
         FILENAME graphviz-ltdl-141d3a21.h
         SHA512 f2d20e849e35060536265f47014c40eb70e57dacd600a9db112fc465fbfa6a66217b44a8c3dc33039c260a27f09d9034b329b03cc28c32a22ec503fcd17b78cd
     )
```
Comment (dg0yt) [inline ports/graphviz/portfile.cmake]: I came to the conclusion that the variable part (`FOLDER_PATH`) can be removed. Just always download the `x64` variant. Both available variants are identical (same SHA512).

Comment (LilyWangLL) [reply ports/graphviz/portfile.cmake]: Yes, I test this locally, this file on `x64` and `x86` folder has the same hash. I removed the codes of `FOLDER_PATH`.

File: ports/graphviz/portfile.cmake [hunk 4]
```diff
@@ -26,8 +24,8 @@ vcpkg_list(SET OPTIONS)
 if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
     vcpkg_download_distfile(
         LTDL_H_PATH
-        URLS "https://gitlab.com/graphviz/graphviz-windows-dependencies/-/raw/141d3a21be904fa8dc2ae3ed01d36684db07a35d/${VCPKG_TARGET_ARCHITECTURE}/include/ltdl.h"
-        FILENAME graphviz-ltdl-141d3a21.h
+        URLS "https://gitlab.com/graphviz/graphviz-windows-dependencies/-/raw/master/x64/include/ltdl.h"
+        FILENAME graphviz-ltdl-master.h
```
Comment (dg0yt) [inline ports/graphviz/portfile.cmake:28]: Maybe we need to collect $5 for each commit which is known to cause the next baseline regression.

Do not use `master`. It is a branch, it content changes, and so the SHA512 changes. Go back to the commit sha or a tag.

**Review by Cheney-W — DISMISSED** [review]

**Comment by dg0yt** [comment]
~~~
error: Failed to download from mirror set
error: https://vcpkgassetcacheeastasia.blob.core.windows.net/cache/adf19357edc62681a2515e7210a752b0e09214b6ce69024e60150e0780059c08a9ab5a162a0562dbc37127438783a24bcde1adb88b559bc95ff9a5bea0eb8b39?*** SECRET ***: curl failed to download with exit code 22
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

curl: (22) The requested URL returned error: 404 The specified blob does not exist.
~~~
The asset is not cached yet, and
~~~
error: https://earth-info.nga.mil/php/download.php?file=wgs-mastertgz: curl failed to download with exit code 6
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:--  0:00:09 --:--:--     0curl: (6) Could not resolve host: earth-info.nga.mil
~~~
Maybe some one is censoring DNS lookups for earth-info.nga.mil?
~~~
$ nslookup earth-info.nga.mil
[...]
Non-authoritative answer:
Name:   earth-info.nga.mil
Address: 214.28.170.173
Name:   earth-info.nga.mil
Address: 214.28.196.146
~~~

**Comment by BillyONeal** [comment]
Is this thing even still maintained?

**Comment by dg0yt** [comment]
> Is this thing even still maintained?

I think yes. The last update is from 2022. This is fairly new in geo terms. vcpkg is still adding ports which are much more outdated.

**Comment by dg0yt** [comment]
> [revert changes of geotrans](https://github.com/microsoft/vcpkg/pull/31772/commits/59eb8157db0e16c02503071726881aac325bccf9)

Why? We know that the source is newer, and we can test it locally. The "breaking" change for CI is enabling the build in `ci.baseline.txt`.

**Comment by LilyWangLL** [comment]
> > [revert changes of geotrans](https://github.com/microsoft/vcpkg/pull/31772/commits/59eb8157db0e16c02503071726881aac325bccf9)
> 
> Why? We know that the source is newer, and we can test it locally. The "breaking" change for CI is enabling the build in `ci.baseline.txt`.

I will submit a new PR to update `geotrans`, let other ports merged first.

**Comment by dg0yt** [comment]
Is this still pending?

File: ports/geotrans/portfile.cmake [hunk 5]
```diff
@@ -4,8 +4,8 @@ set(VCPKG_LIBRARY_LINKAGE "dynamic")
 # which we re-build anyway.  There is no source only package provided or it would be preferred (and smaller).
 vcpkg_download_distfile(ARCHIVE
     URLS "https://earth-info.nga.mil/php/download.php?file=wgs-mastertgz"
-    FILENAME "geotrans-3.9-master.tgz"
-    SHA512 b0bd5ca6eb584eebf0adc89eb86ec949f372bb4ea92c559f74f2055b0f121369b3d8de0bcff7b7db9abc57ac5a129d5c283d67f098e4af12d91b41747a76e541
+    FILENAME "geotrans-${VERSION}-master-b0bd5ca.tgz"
```
Comment (BillyONeal) [inline ports/geotrans/portfile.cmake:7]: I don't think there's a point in using `${VERSION}` when there's something that looks like a commit sha in there.

Comment (LilyWangLL) [reply ports/geotrans/portfile.cmake]: I changed it, now only add the current hash value.
```
geotrans-3.9-master-adf1935.tgz
```

**Review by FrankXie05 — APPROVED** [review]

File: ports/graphviz/portfile.cmake [hunk 6]
```diff
@@ -26,7 +24,7 @@ vcpkg_list(SET OPTIONS)
 if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
     vcpkg_download_distfile(
         LTDL_H_PATH
-        URLS "https://gitlab.com/graphviz/graphviz-windows-dependencies/-/raw/141d3a21be904fa8dc2ae3ed01d36684db07a35d/${VCPKG_TARGET_ARCHITECTURE}/include/ltdl.h"
+        URLS "https://gitlab.com/graphviz/graphviz-windows-dependencies/-/raw/141d3a21be904fa8dc2ae3ed01d36684db07a35d/x64/include/ltdl.h"
```
Comment (JavierMatosD) [inline ports/graphviz/portfile.cmake:27]: Is this not available for other target architectures? It looks like GitLab is down at the moment.

**Review by JavierMatosD — COMMENTED** [review]
Setting the PR to "draft." Please set it to "ready for review" once you've responded.

File: ports/graphviz/portfile.cmake (hunk 6 above)
Comment (BillyONeal) [reply ports/graphviz/portfile.cmake:27]: It looks like the header is not architecture specific to me.

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**